javascript/javascript3/week1/weather/index.js

```diff
@@ -0,0 +1,102 @@
+const iconElement = document.querySelector(".weather-icon");
+const tempElement = document.querySelector(".temperature-value p");
+const descElement = document.querySelector(".temperature-description p");
+const locationElement = document.querySelector(".location p");
+const notificationElement = document.querySelector(".notification");
+
+// date
+n = new Date();
+y = n.getFullYear();
+m = n.getMonth() + 1;
+d = n.getDate();
+document.getElementById("date").innerHTML = "Today Forecast " + d + "/" + m + "/" + y;
+
+
+const weather = {};
+
+weather.temperature = {
+    unit: "celsius"
+}
+
+const KELVIN = 273;
+
+const key = '03cb5d467f485a1ccf2fca9e32555f53';
+//getting geolocation 
+
+if ('geolocation' in navigator) {
+    navigator.geolocation.getCurrentPosition(setPosition, showError);
+} else {
+    notificationElement.style.display = "block";
+    notificationElement.innerHTML = "<p>Browser doesn't Support Geolocation</p>";
+}
+function setPosition(position) {
+    let latitude = position.coords.latitude;
+    let longitude = position.coords.longitude;
+
+    getWeather(latitude, longitude);
+}
+function showError(error) {
+    notificationElement.style.display = "block";
+    notificationElement.innerHTML = `<p> ${error.message} </p>`;
+}
+
+//fetching data based on current location
+
+function getWeather(latitude, longitude) {
+    let api = `http://api.openweathermap.org/data/2.5/weather?lat=${latitude}&lon=${longitude}&appid=${key}`;
+
+    fetch(api)
+        .then(function (response) {
+            let data = response.json();
+            return data;
+        })
+        .then(function (data) {
+            console.log(data);
+            weather.temperature.value = Math.floor(data.main.temp - KELVIN);
+            weather.description = data.weather[0].description;
+            weather.iconId = data.weather[0].icon;
+            weather.city = data.name;
+            weather.country = data.sys.country;
+            weather.windspeed = data.wind.speed;
+            weather.sunrise = data.sys.sunrise;
+            weather.sunset = data.sys.sunset;
+        })
+        .then(function () {
+            displayWeather();
+        });
```

Comment:
No need to new `then` because this callback function does not return a new promise
```js
...
weather.sunset = data.sys.sunset;
displayWeather();
```
- In general, try to avoid side effects by using global variables, here you have weather object as a global variable. it is better to have a local one and then pass it as a parameter to `displayWeather` function

You can keep it for now, but keep in mind to avoid this way for better code. Then you will insure that not other function can effect the value of `weather` object